### PR TITLE
Promote Immutable Secrets/ConfigMaps to Beta

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/enhancements/issues/1412

### CODE REVIEW

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-21189--kubernetes-io-vnext-staging.netlify.app/docs/concepts/configuration/secret/
https://deploy-preview-21189--kubernetes-io-vnext-staging.netlify.app/docs/reference/command-line-tools-reference/feature-gates/#feature-gates-for-alpha-or-beta-features

**Comment by sftim** [comment]
Aside: let's document immutable ConfigMaps on https://kubernetes.io/docs/concepts/configuration/configmap/

**Comment by wojtek-t** [comment]
> Aside: let's document immutable ConfigMaps on https://kubernetes.io/docs/concepts/configuration/configmap/

We should basically copy-paste this whole section to configmaps: https://kubernetes.io/docs/concepts/configuration/secret/#mounted-secrets-are-updated-automatically
[This works exactly the same way for secrets and configmaps and has never been added.]

@sftim - should I send a separate PR to add that (this is more than just immutable secrets/configmaps feature-related stuff)? If so, to what branches?

**Comment by sftim** [comment]
> @sftim - should I send a separate PR to add that (this is more than just immutable secrets/configmaps feature-related stuff)? If so, to what branches?

The best thing I think would be a separate GitHub issue and then a PR against _master_ @wojtek-t. I'd be very happy to review that.

**Comment by wojtek-t** [comment]
@sftim - yes - #1407 is exactly that. Let me sent a PR for it.

**Comment by wojtek-t** [comment]
@sftim - opened https://github.com/kubernetes/website/pull/21452

**Comment by wojtek-t** [comment]
Simple rebase - (just a conflict in feature gates file).

@sftim - PTAL

**Comment by sftim** [comment]
Sure
/lgtm

**Comment by sftim** [comment]
https://github.com/kubernetes/website/blob/dev-1.19/content/en/docs/concepts/configuration/configmap.md exists now @wojtek-t so _I think_ the right approach here is rebase against _dev-1.19_ and then update `content/en/docs/concepts/configuration/configmap.md` as well.

Does that seem right?

**Comment by wojtek-t** [comment]
@sftim - done; can you please re-lgtm?

**Comment by wojtek-t** [comment]
@sftim - friendly ping

**Comment by wojtek-t** [comment]
@kbhawkey - can you please approve?

File: content/en/docs/concepts/configuration/configmap.md [hunk 1]
```diff
@@ -212,9 +212,9 @@ when new keys are projected to the Pod can be as long as the kubelet sync period
 propagation delay, where the cache propagation delay depends on the chosen cache type
 (it equals to watch propagation delay, ttl of cache, or zero correspondingly).
 
-{{< feature-state for_k8s_version="v1.18" state="alpha" >}}
+{{< feature-state for_k8s_version="v1.18" state="beta" >}}
```
Comment (kbhawkey) [inline content/en/docs/concepts/configuration/configmap.md:215]: Hi @wojtek-t .
For ConfigMap, should the k8s version move to `v1.19`?

**Comment by kbhawkey** [comment]
Another page preview:
https://deploy-preview-21189--kubernetes-io-vnext-staging.netlify.app/docs/concepts/configuration/configmap/

**Comment by sftim** [comment]
https://github.com/kubernetes/website/pull/21189#pullrequestreview-438396666 is a good catch
/lgtm cancel

File: content/en/docs/concepts/configuration/configmap.md (hunk 1 above)
Comment (wojtek-t) [reply content/en/docs/concepts/configuration/configmap.md]: good catch -done

**Comment by wojtek-t** [comment]
@sftim @kbhawkey - PTAL

**Pull request merged**